distributed/nanny.py

```diff
@@ -44,7 +43,8 @@ def __init__(self, scheduler_ip=None, scheduler_port=None,
                  name=None, memory_limit='auto', reconnect=True,
                  validate=False, quiet=False, resources=None, silence_logs=None,
                  death_timeout=None, preload=(), preload_argv=[], security=None,
-                 contact_address=None, listen_address=None, **kwargs):
+                 contact_address=None, listen_address=None, worker_class=None,
```

Comment:
Maybe we can set `worker_class=Worker` to avoid the check later?

I'm +1 regardless though if you want to merge @martindurant

Reply:
The current line fits in better with the style of the other optional arguments, so I would keep it this way. To me, None says something like "do not worry about this parameter, unless you specifically know you need it".

Reply:
I don't think that that's convention across the ecosystem.  I think that default values are usually used rather than None.

Reply:
None is typically used when a default value can't be known ahead of time because it depends on other arguments passed in.  In this case we know what the default value is, `Worker`, and so it's nice to specify it explicitly.  This both reduces arg handling code later on and also provides the user a suggestion in documentation and docstrings.

Reply:
It is certainly common :) particularly in very long lists like this, where one suspects it might eventually be replaced by `**kwargs` .  Thinking https://pandas.pydata.org/pandas-docs/stable/generated/pandas.read_csv.html

Reply:
Sure, that looks like a fine example to look at.  Is there a keyword in that list that has a known default value that isn't listed as a default?

Reply:
To be sure, this was nothing more than a gut decision, I don't mind either way.

(encoding, dialect, line_terminator, float_precision, date_parser ...)

Reply:
Looking at encoding as an example I'm unable to find a line like 

```python
if encoding is None:
    encoding = 'foo'
```

Instead there is logic in there

```python
def _read(filepath_or_buffer, kwds):
    """Generic reader of line files."""
    encoding = kwds.get('encoding', None)
    if encoding is not None:
        encoding = re.sub('_', '-', encoding).lower()
        kwds['encoding'] = encoding
```

I suspect, but have not verified, that the same is true with the others as well.